pkg/controller/kieapp/defaults/upgrade.go

```diff
@@ -49,3 +52,70 @@ func MajorMinorMicro(productVersion string) (major, minor, micro string) {
 	}
 	return version[0], version[1], version[2]
 }
+
+// getConfigVersionDiffs ...
+func getConfigVersionDiffs(fromVersion, toVersion string, service api.PlatformService) (err error) {
```

Comment:
This uses a named return variable, so you should just set the var and `return`... or don't name it and `return nil` or value as appropriate. It's confusing this way, though not incorrect as I can see implemented.